markdown/developers/adding_modules.md

```diff
@@ -523,6 +523,91 @@ If the module absolute cannot run using tiny test data, there is a possibility t
 nextflow run tests/modules/<nameofmodule> -entry test_<nameofmodule> -c tests/config/nextflow.config -stub-run
 ```
 
+## What is the `meta` map?
+
+In nf-core DSL2 pipelines, to add sample information and metadata, we use a meta variable. This avoids the need to create separate channels for each new characteristic. 
+The meta variable can be passed down to processes as a tuple of the channel containing the actual samples, e.g. FastQ files, and the meta variable. 
+The `meta map` is a [groovy map](https://www.tutorialspoint.com/groovy/groovy_maps.htm), which is like a python dictionary.
```

Comment:
Should we show the whole channel structure at the beginning before introducing only the map e.g. `[[id: 'test', single_end: false], file]` explaining that the first item is the map and the second the file(s) itself or would it be more confusing?

Reply:
```suggestion
The `meta map` is a [groovy map](https://www.tutorialspoint.com/groovy/groovy_maps.htm), which is like a python dictionary, as shown below:
```

Reply:
> Should we show the whole channel structure at the beginning before introducing only the map e.g. `[[id: 'test', single_end: false], file]` explaining that the first item is the map and the second the file(s) itself, or would it be more confusing?

Hmm, it could work, could you write out what you're thinking? I don't know what the difference between the two would be.